pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -480,6 +481,7 @@ public synchronized void readEntriesComplete(List<Entry> entries, Object ctx) {
             // All consumers got disconnected before the completion of the read operation
             entries.forEach(Entry::release);
             cursor.rewind();
+            clearDelayedMessages();
```

Comment:
If `clearDelayedMessages()` binds with `cursor.rewind()`, it's better to wrap them into one method.